update_factorio.py

```diff
@@ -138,11 +138,28 @@ def verbose_aware_exec(exec_args, verbose=False):
         raise
 
 
+def get_username_token(factoriopath):
+    with open(os.path.normpath(factoriopath + '/../../../player-data.json')) as data_file:  
```

Comment:
Also, the `player-data.json` may be located somewhere in the user profile directory (installer version and Steam both do this), so you can't just go searching for it from the binary. You'll need to read the (.ini, I think?) file that specifies where user data lives, which _is_ in a known path relative to the Factorio executable.